### PR TITLE
Add IAM SAML Federations, Federation Certificates

### PR DESCRIPTION
**Pull request by Icerzack**
Changes: 

* add SAML Federations, Federation Certificates

### CODE REVIEW

File: selectel/resource_selectel_iam_saml_federation_certificate_v1.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+func resourceIAMSAMLFederationCertificateV1Delete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	iamClient, diagErr := getIAMClient(meta)
+	if diagErr != nil {
+		return diagErr
+	}
+
+	log.Print(msgDelete(objectSAMLFederationCertificate, d.Id()))
+	err := iamClient.SAMLFederations.Certificates.Delete(ctx, d.Get("federation_id").(string), d.Id())
+	if err != nil && !errors.Is(err, iamerrors.ErrFederationNotFound) {
```
Comment (saydamir) [inline selectel/resource_selectel_iam_saml_federation_certificate_v1.go:148]: Should we handle “certificate not found” error here?

**Review by saydamir — CHANGES_REQUESTED** [review]

File: website/docs/r/iam_saml_federation_v1.html.markdown [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+* `description` - (Optional) Federation description.
+
+* `issuer` - (Required) ID of the credential provider.
+
+* `sso_url` - (Required) Link to the credential provider login page.
+
+* `sign_authn_requests` - (Required) Enable authentication requests signing.
+
+* `forec_authn` - (Optional) Require users to authenticate via SSO every time they log in.
```
[the remark this answers is not in the document]
Comment (Icerzack) [reply website/docs/r/iam_saml_federation_v1.html.markdown]: Thanks, fixed it

File: selectel/resource_selectel_iam_saml_federation_v1.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+				Description: "IdP server endpoint URL.",
+			},
+			"sso_url": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Single sign-on endpoint URL.",
+			},
+			"sign_authn_requests": {
+				Type:        schema.TypeBool,
+				Required:    true,
```
[the remark this answers is not in the document]
Comment (Icerzack) [reply selectel/resource_selectel_iam_saml_federation_v1.go]: Thank you, changed to `Optional` with default `false` value

File: selectel/resource_selectel_iam_saml_federation_v1.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: "Description of the Federation.",
+			},
+			"issuer": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "IdP server endpoint URL.",
```
[the remark this answers is not in the document]
Comment (Icerzack) [reply selectel/resource_selectel_iam_saml_federation_v1.go]: Fixed

File: selectel/resource_selectel_iam_saml_federation_v1.go [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+	}
+}
+
+func resourceIAMSAMLFederationV1Create(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	iamClient, diagErr := getIAMClient(meta)
+	if diagErr != nil {
+		return diagErr
+	}
+
+	log.Print(msgCreate(objectSAMLFederation, d.Id()))
```
[the remark this answers is not in the document]
Comment (Icerzack) [reply selectel/resource_selectel_iam_saml_federation_v1.go]: Didn't notice, moved it down and printed `opts`

File: selectel/resource_selectel_iam_saml_federation_certificate_v1.go (hunk 1 above)
Comment (Icerzack) [reply selectel/resource_selectel_iam_saml_federation_certificate_v1.go]: Unfortunately, there is no `ErrFederationCertificateNotFound`, so I decided to print any error

Comment (saydamir) [reply selectel/resource_selectel_iam_saml_federation_certificate_v1.go]: I think we should add this error to iam-go, to handle such situations in a consistent way.

File: website/docs/r/iam_saml_federation_v1.html.markdown [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+* `name` - (Required) Federation name.
+
+* `description` - (Optional) Federation description.
+
+* `issuer` - (Required) ID of the credential provider.
+
+* `sso_url` - (Required) Link to the credential provider login page.
+
+* `sign_authn_requests` - (Required) Enable authentication requests signing.
```
Comment (saydamir) [inline website/docs/r/iam_saml_federation_v1.html.markdown:39]: Still required here

File: website/docs/r/iam_saml_federation_v1.html.markdown [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+## Example Usage
+
+```hcl
+resource "selectel_iam_saml_federation_v1" "federation_1" {
+  name                  = "federation name"
+  description           = "simple description"
+  issuer                = "http://localhost:8080/realms/master"
+  sso_url               = "http://localhost:8080/realms/master/protocol/saml"
+  sign_authn_requests   = true
+  force_authn           = true
```
Comment (saydamir) [inline website/docs/r/iam_saml_federation_v1.html.markdown:24]: Is it common practice to list all available options in an example?
I would either set default values for these two fields or remove them from the example.

Comment (Icerzack) [reply website/docs/r/iam_saml_federation_v1.html.markdown]: Agree, I removed these optional params, only description left

File: selectel/resource_selectel_iam_saml_federation_certificate_v1.go (hunk 1 above)
Comment (Icerzack) [reply selectel/resource_selectel_iam_saml_federation_certificate_v1.go]: Done!

**Review by saydamir — APPROVED** [review]

**Review by shlykov — APPROVED** [review]

**Review by shlykov — APPROVED** [review]

**Pull request merged**